pkg/providers/aws/credentials_sts_manager.go

```diff
@@ -35,12 +35,9 @@ func NewSTSCredentialManager(client client.Client) *STSCredentialManager {
 }
 
 //ReconcileProviderCredentials Ensure the credentials the AWS provider requires are available
-func (m *STSCredentialManager) ReconcileProviderCredentials(ctx context.Context, ns string) (*Credentials, error) {
-	secret, err := getSTSCredentialsSecret(ctx, m.Client, m.OperatorNamespace)
-	if err != nil {
-		return nil, errorUtil.Wrapf(err, "failed to get aws STS credentials secret %s", defaultSTSCredentialSecretName)
-	}
-
+func (m *STSCredentialManager) ReconcileProviderCredentials(ctx context.Context, _ string) (*Credentials, error) {
+	// ignoring the error here as this is validated when calling NewCredentialManager()
+	secret, _ := getSTSCredentialsSecret(ctx, m.Client, m.OperatorNamespace)
```

Comment:
I think it might be better to still keep the error check as while unlikely, it can hit a nil pointer due to the error being ignored :thinking: (e.g. secret was there during NewCredentialManager, but was unable to get the secret during this function for some reason )